bigflow/dataproc.py

```diff
@@ -0,0 +1,358 @@
+import base64
+import functools
+import datetime
+import time
+import inspect
+import io
+import os
+import pathlib
+import pickle
+import random
+import re
+import string
+import subprocess
+import textwrap
+import time
+import typing
+import contextlib
+import logging
+
+from google.cloud import dataproc_v1, storage
+
+import bigflow.configuration
+import bigflow.resources
+import bigflow.commons
+
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    'PySparkJob',
+]
+
+DEFAULT_REQUIREMENTS = [
+    'bigflow[dataproc,log,bigquery]==1.0.dev67',
+]
+
+
+class PySparkJob:
+
+    def __init__(
+        self,
+        id,
+        driver_callable: typing.Callable,
+        bucket_id: str,
+        gcp_project_id: str,
+        gcp_region: str,
+        driver_arguments: typing.Optional[dict] = None,
+        pip_packages: typing.Union[typing.Iterable[str], pathlib.Path] = (),
+        jar_file_uris: typing.Iterable[str] = ('gs://spark-lib/bigquery/spark-bigquery-latest_2.12.jar',),
+        worker_num_instances: int = 2,
+        worker_machine_type: str = 'n1-standard-1',
+        env: str = '',
+        setup_file: str = '',
```

Comment:
Why an empty string instead of None?

Reply:
Agree, better to use Optional[str] = None, fixed.